src/activity/iActivity.inl.hxx

```diff
@@ -2,7 +2,18 @@
 template <typename StateType, typename... Args>
 StateType & iActivity::createState(Args &&... args)
 {
-  StateType* state = new StateType(std::forward(args)..., GetService<UILoopMQ>());
+  StateType* state = new StateType(std::forward<Args>(args)..., GetService<UILoopMQ>());
   m_States.emplace_back(state);
```

Comment:
I think there is a memory leak issue here. m_States vector store pointer to the StateType (e.g. ButtonState). During the call of default destructor only vector and these pointers will be destroyed, but not the objects on the heap. I think in iActivity destructor, the destructor for each object on heap must be called explicitly.